OfficeDevPnP.Core/OfficeDevPnP.Core/Framework/Provisioning/Model/PropertyBagEntry.cs

```diff
@@ -1,41 +1,19 @@
-﻿using System;
+﻿using OfficeDevPnP.Core.Framework.Provisioning.Model.Attributes;
+using OfficeDevPnP.Core.Framework.Provisioning.Model.Comparers;
+using OfficeDevPnP.Core.Framework.Provisioning.Model.HashFormatters;
+using System;
 
 namespace OfficeDevPnP.Core.Framework.Provisioning.Model
 {
-    public class PropertyBagEntry : IEquatable<PropertyBagEntry>
+    public class PropertyBagEntry : ModelBase<PropertyBagEntry>
     {
-        #region Properties
-
+        [HashCodeIdentifier]
         public string Key { get; set; }
-            
+        [HashCodeIdentifier]
         public string Value { get; set; }
-
-        #endregion
-
-        #region Comparison code
-
-        public override int GetHashCode()
+        public PropertyBagEntry()
         {
```

Comment:
Not terribly in love with the idea of always overriding the base constructor.. Could mark it virtual to signify intent.